### ISSUE TITLE
figure out why test_multiprocessing is being skipped unnecessarily

### ISSUE THREAD

**Issue by mperrin** [mperrin/poppy#130]
Moving this from email to a github issue: 

@mperrin wrote:
So I am double checking through the test suite, and I’ve noticed that the test_multiprocessing function is getting skipped even when we are using Python 3.4, both on Travis and on my local machine.  See output below. 

The code to check version and skip looks fine: 
```
    @pytest.mark.skipif( (sys.version_info < (3,4,0) ),
            reason="Python 3.4 required for reliable forkserver start method")
```

I’m testing on python 3.4.3 so I don’t understand why this is getting skipped. I verified that sys.version_info < (3,4,0)  is False, so I’m stumped. Any ideas? 

@josePhoenix replied:
I think it's being skipped because you don't have the pytest package installed in your 3.4 environment. Probably the same thing on Travis. I'm surprised it shows up as skipped, when if it were executing as a "real" script it would never enter the "if _HAVE_PYTEST:" block in test_multiprocessing.py.

Wait, no, it's the 'remote_data' helper… I think…

**Comment by mperrin** [comment]
I think the problem is definitely something with the `@remote_data` decorator.

With the `@remote_data` helper commented out, the multiprocessing test flat-out fails rather than is skipped: The astropy `disable_internet` test customization crushes it mercilessly for the attempted local socket access. See https://github.com/astropy/astropy/issues/2557  So we do need to include that decorator. 

But clearly I don't understand something about how to use the `@remote_data` decorator properly. I tested the following simple case:
```
       @remote_data 
        def test_test_skipping():
            assert True
```
and this results in that test case being skipped when I run `python setup.py test`.  

@embray can you provide any guidance here? Thanks.

**Comment by embray** [comment]
That shouldn't happen--the issue you're talking about was fixed in Astropy v1.0.3.

**Comment by embray** [comment]
astropy/astropy#3713

**Comment by mperrin** [comment]
OK, it turns out my desktop was still on astropy 1.0.2, so I updated that. And then I found that I could remove the `@remote_data` decorator and the test would still be allowed to run, so that's great. The Travis test run with that change is going now. Fingers crossed it will all pass.

I'm still a little confused why adding that decorator makes Travis skip running the test entirely (and I checked that Travis is using the 1.0.5 astropy conda package). If we were to have some future test that relies on some remote data functionality (which isn't totally out of the question eventually) is there some other step necessary so that such tests can run on Travis?

Thanks.
